test/tests/VehicleTest.cpp

```diff
@@ -0,0 +1,1218 @@
+    assert(false);
+    return false;
+}
+
+bool Peep::SetName(std::string_view value)
+{
+    // Probably overkill
+    assert(false);
+    return true;
+}
+
+template<> bool SpriteBase::Is<Peep>() const
+{
+    // Probably overkill
+    assert(false);
+    return sprite_identifier == SpriteIdentifier::Peep;
+}
+
+template<> bool SpriteBase::Is<Guest>() const
+{
+    // Probably overkill
+    assert(false);
+    auto peep = As<Peep>();
+    return peep && peep->AssignedPeepType == PeepType::Guest;
+}
+
+template<> bool SpriteBase::Is<VehicleCrashParticle>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::CrashedVehicleParticle;
+}
+
+template<> bool SpriteBase::Is<CrashSplashParticle>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::CrashSplash;
+}
+
+template<> bool SpriteBase::Is<JumpingFountain>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc
+        && (misc->SubType == MiscEntityType::JumpingFountainSnow || misc->SubType == MiscEntityType::JumpingFountainWater);
+}
+
+template<> bool SpriteBase::Is<Balloon>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::Balloon;
+}
+
+template<> bool SpriteBase::Is<Duck>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::Duck;
+}
+
+template<> bool SpriteBase::Is<MoneyEffect>() const
+{
+    // Probably overkill
+    assert(false);
+    auto* misc = As<MiscEntity>();
+    return misc && misc->SubType == MiscEntityType::MoneyEffect;
+}
+
+Guest* Peep::AsGuest()
+{
+    // Probably overkill
+    assert(false);
+    return nullptr;
+}
+
+int32_t Guest::GetEasterEggNameId() const
+{
+    // TODO(janisozaur): do something here
+    return -1;
+}
+
+TEST(Vehicle, Create)
+{
+    [[maybe_unused]] Vehicle v{};
+}
+
+TEST(Vehicle, SimpleUpdate)
+{
+    Vehicle v{};
+    v.Update();
+}
+
+TEST(Vehicle, CableLiftUpdate)
+{
+    reset_sprite_list();
+    Ride& r = _rideList[0];
+    r = Ride{};
+    r.id = 0;
+    Vehicle* v = cable_lift_segment_create(r, 0, 0, 0, 0, 0, 0, true);
+    v->ride_subtype = RIDE_ENTRY_INDEX_NULL;
+    v->status = Vehicle::Status::MovingToEndOfStation;
+    v->next_vehicle_on_train = SPRITE_INDEX_NULL;
+    v->Update();
+}
+
+static constexpr const CoordsXY word_9A3AB4[4] = {
+    { 0, 0 },
+    { 0, -96 },
+    { -96, -96 },
+    { -96, 0 },
+};
+
+// clang-format off
+static constexpr const CoordsXY word_9A2A60[] = {
+    { 0, 16 },
+    { 16, 31 },
+    { 31, 16 },
+    { 16, 0 },
+    { 16, 16 },
+    { 64, 64 },
+    { 64, -32 },
+    { -32, -32 },
+    { -32, 64 },
+};
+// clang-format on
+
+static Vehicle* vehicle_create_car(
```

Comment:
Copied verbatim from `Ride.cpp`, but including that file in the new target pulls a whole lot of other dependencies. I haven't really paid much attention to what it does, might be simplified